### ISSUE TITLE
[terra-demographics-banner] Remove photo prop

### ISSUE THREAD

**Issue by benbcai** [cerner/terra-core#3002]
# Feature Request

## Description
The `photo` prop for showing an image on the demographics banner is being replaced by an avatar in #2774. Both `photo` and avatar are supported with avatar taking precedence. The `photo` prop should be removed and completed replaced by avatar in the next MVB.
